common/chat.ts

```diff
@@ -15,10 +15,20 @@ export enum ClientChatMessageType {
 
 export type ChatMessageType = ServerChatMessageType | ClientChatMessageType
 
-// TODO(2Pac): Include more information here, e.g. channel permissions, join date, etc.
+export interface ChannelPermissions {
+  kick: boolean
+  ban: boolean
+  changeTopic: boolean
+  togglePrivate: boolean
+  editPermissions: boolean
+  owner: boolean
+}
+
 export interface ChatUser {
   id: SbUserId
   name: string
+  joinDate: number
```

Comment:
I'm not a fan of adding these to this structure. If we want to include chat-specific information then the `name` should be removed and this structure should only contain a userId and these chat-specific things, and the structure with id + name should be sent separately from this structure